packages/agents-hosting-extensions-teams/src/teamsAgentExtension.ts

```diff
@@ -30,7 +31,20 @@ export class TeamsAgentExtension<TState extends TurnState = TurnState> extends A
     return this._taskModule
   }
 
-  onMessageEdit = (handler: RouteHandler<TurnState>) => {
+  onFeedback (handler: RouteHandler<TurnState>) {
+    const routeSel: RouteSelector = (context: TurnContext) => {
+      return Promise.resolve(
+        context.activity.type === ActivityTypes.Invoke &&
+        context.activity.channelId === 'msteams' &&
+        context.activity.name === 'message/submitAction' &&
+        (context.activity.value as FeedbackLoopData).actionName === 'feedback'
```

Comment:
consider adding a zod schema for feedbackLoopData and parse here, instead of the casting.